Update docs in prep for 1.6.1 (#133)
aws/aws-encryption-sdk-java: CHANGELOG.md

# Changelog

## 1.6.1 -- Unreleased

### Deprecation Warnings

> * `AwsCrypto.encryptString()` and `AwsCrypto.decryptString()` are now deprecated.

Reviewer: Since this is a changelog, we should use the declarative.

```suggestion
* Deprecated `AwsCrypto.encryptString()` and `AwsCrypto.decryptString()`.
```

>   Please move to `AwsCrypto.encryptData()` and `AwsCrypto.decryptData()` with manual Base64 encoding/decoding.

Reviewer: ```suggestion
Replace your calls to these methods with calls to  `AwsCrypto.encryptData()` and `AwsCrypto.decryptData()`. These methods use manual Base64 encoding/decoding, so they are fully compatible with other language implementations of the AWS Encryption SDK.
```

Author: The new methods don't use Base64 encoding and will require the developer to apply it themselves. Do you have better wording for that?

Reviewer: Replace your calls to these methods with calls to  `AwsCrypto.encryptData()` and `AwsCrypto.decryptData()`. Because these methods don't perform any Base64 encoding or decoding, they are fully compatible with other language implementations of the AWS Encryption SDK.

Reviewer: Replace your calls to these methods with calls to AwsCrypto.encryptData() and AwsCrypto.decryptData(). Unlike the deprecated methods, these methods don't perform any Base64 encoding or decoding, so they are fully compatible with other language implementations of the AWS Encryption SDK.

If you need Base64 encoding or decoding for your application, you can add it outside of the AWS Encryption SDK.

  [PR #120](https://github.com/aws/aws-encryption-sdk-java/pull/120)

### Patches
* Correctly validate version [PR #116](https://github.com/aws/aws-encryption-sdk-java/pull/116)

> * `ParsedCiphertext` now properly handles truncated input [PR #119](https://github.com/aws/aws-encryption-sdk-java/pull/119)

Reviewer: ```suggestion
* `ParsedCiphertext` now handles truncated input properly [PR #119](https://github.com/aws/aws-encryption-sdk-java/pull/119)
```

### Maintenance

> * Add support for standard test vectors via `testVectorZip` system property. [PR #127](https://github.com/aws/aws-encryption-sdk-java/pull/127)

Reviewer: s/Add/Added

This is a done deal, not something in progress.

> * Remove all explicit cryptographic dependencies on BouncyCastle. PRs

Reviewer: ```suggestion
* Remove all explicit cryptographic dependencies on BouncyCastle. The AWS Encryption SDK for Java still uses Bouncy Castle for other tasks. PRs
```
